### PR TITLE
Improve Exception Handling for Invalid Sort Fields

### PR DESCRIPTION
**Pull request by Guhapriya01**
**Changes Made:**

- Refactored `PropertyReferenceException` to throw a meaningful error response across all the service methods that deals with pagination and sorting.

**Error Response Example:**

```json
{
    "timestamp": "2024-10-03T07:37:08.364+00:00",
    "message": "The specified 'sortBy' value is invalid.",
    "details": "/api/books"
}
```

### CODE REVIEW

File: src/main/java/com/libraryman_api/borrowing/BorrowingService.java [hunk 1]
```diff
@@ -66,9 +68,14 @@ public BorrowingService(BorrowingRepository borrowingRepository, FineRepository
      *
      * @param pageable the pagination information, including the page number and size
      * @return a {@link Page} of {@link Borrowings} representing all borrowings
+     * @throws InvalidSortFieldException if an invalid sortBy field is specified
      */
     public Page<Borrowings> getAllBorrowings(Pageable pageable) {
```
Comment (ajaynegi45) [inline src/main/java/com/libraryman_api/borrowing/BorrowingService.java:73]: What sorting parameters can be passed here?

File: src/main/java/com/libraryman_api/borrowing/BorrowingService.java [hunk 2]
```diff
@@ -264,14 +271,19 @@ private BigDecimal calculateFineAmount(Borrowings borrowing) {
...
      * @throws ResourceNotFoundException if the member has not borrowed any books
+     * @throws InvalidSortFieldException if an invalid sortBy field is specified
      * @return a {@link Page} of {@link Borrowings} representing all borrowing associated with a specific member
      */
-    public Page<Borrowings> getAllBorrowingsOfMember(int memberId, Pageable pageable) {        
-        Page<Borrowings> borrowings = borrowingRepository.findByMember_memberId(memberId, pageable);
-        
-        if (borrowings.isEmpty()) {
-            throw new ResourceNotFoundException("Member didn't borrow any book");
+    public Page<Borrowings> getAllBorrowingsOfMember(int memberId, Pageable pageable) {
```
Comment (ajaynegi45) [inline src/main/java/com/libraryman_api/borrowing/BorrowingService.java:277]: What sorting parameters can be passed here?

File: src/main/java/com/libraryman_api/member/MemberService.java [hunk 3]
```diff
@@ -46,9 +48,14 @@ public MemberService(MemberRepository memberRepository, NotificationService noti
      *
      * @param pageable the pagination information, including the page number and size
      * @return a {@link Page} of {@link Members} representing all members
+     * @throws InvalidSortFieldException if an invalid sortBy field is specified
      */
     public Page<Members> getAllMembers(Pageable pageable) {
-        return memberRepository.findAll(pageable);
+        try {
```
Comment (ajaynegi45) [inline src/main/java/com/libraryman_api/member/MemberService.java:54]: What sorting parameters can be passed here?

**Review by ajaynegi45 — APPROVED** [review]
Hi @Guhapriya01,

Thank you for your prompt work on this issue. I appreciate your dedication to the project.

I have reviewed your work, and I'm pleased to inform you that I have merged your pull request. Your changes looks fantastic, and it aligns perfectly with the project's design and usability goals. Great job!

I have mention some comment reply them.

If you have any more contributions or if there's anything else you'd like to work on in the future, please don't hesitate to let us know. Your efforts are valuable to the project's success.

Once again, thank you for your contribution! ❤️

File: src/main/java/com/libraryman_api/member/MemberService.java (hunk 3 above)
Comment (Guhapriya01) [reply src/main/java/com/libraryman_api/member/MemberService.java:54]: For **Members**, any field in the `Members` model (e.g., name, email, etc.) can be used for sorting.

File: src/main/java/com/libraryman_api/borrowing/BorrowingService.java (hunk 2 above)
Comment (Guhapriya01) [reply src/main/java/com/libraryman_api/borrowing/BorrowingService.java:277]: For **Borrowings**, all fields in the `Borrowings` model (e.g., borrowingId, etc.) are available for sorting. If you need to sort by nested fields (like Member.name), use the format `fieldName_propertyName` (e.g., member_name, book_name).

File: src/main/java/com/libraryman_api/borrowing/BorrowingService.java (hunk 1 above)
Comment (Guhapriya01) [reply src/main/java/com/libraryman_api/borrowing/BorrowingService.java:73]: For **Books**, any field in the `Book` model (e.g., title, publisher, author, etc.) can be used for sorting.

**Comment by Guhapriya01** [comment]
Hi Ajay!

Thank you for the kind words! I’m glad to hear that my changes were well-received and aligned with the project goals. I have replied to the comments you mentioned and am eager to keep contributing to the project!

Thanks again!

**Pull request merged**